### ISSUE TITLE
Accessibility Issue - Buttons with identical accessible names

### ISSUE THREAD

**Issue by Bradrajkumar** [microsoft/BotFramework-WebChat#4470]
**Issue Description : -**

Interactive elements need a programmatically determined name, role, and value. These properties must also be accurate and appropriate.
There are buttons that have identical accessible names. This may particularly disadvantage users that interface with the accessibility properties of elements, such as screen reader and voice recognition users. That the accessible name is identical may confuse or mislead users.
This has been raised as an Additional Issue as the accessible names are not wrong. However, the accessible names could be made more explicit, such that they are easier to understand and differentiate.

**Success Criteria : -**

Make sure that interactive elements have unique and descriptive names. In this case, make sure that both buttons include a reference to the specific product.
This can be achieved by using the id of the container for the product name and an aria-describedby attribute on the buttons that references this. Ensure that the id and aria-describedby values match. For example:

<div role="heading" aria-level="3" id="context"...>
<p >some text</p>
</div>
<button aria-describedby="context">Select this product</button>

![6_1](https://user-images.githubusercontent.com/34365788/198262151-f0a7830a-3a54-4506-9179-910170c63b01.png)
![6_2](https://user-images.githubusercontent.com/34365788/198262157-93e0bcaa-21e4-4d37-b71e-8ff6a660250e.png)

**Steps to reproduce**

1. Observe the interactive elements.
2. Inspect the elements and review the DOM.
3. Review the computed accessibility properties.
4. Observe that the names are programmatically determined.
5. Observe that the names are identical.

**Webchat Version :- 4.13.0**

**Comment by compulim** [comment]
> To DRI: This is an issue on Adaptive Cards side. AC is lacking accessibility features to associate the button with other elements in the card.

If I understand this correctly, there are two ways recommended by WAI-ARIA:

- Use [`aria-controls` property](https://w3c.github.io/aria/#aria-controls) on the button to identify it is being associated with the product image element
- Use [`aria-description`](https://w3c.github.io/aria/#aria-description) or `aria-describedby` to further enhance the content of the button
   - The enhancement would include the product name, such as "Select Orange"

However, this is under an element with class of `ac-adaptiveCard`. That means, this is owned by Adaptive Cards team.

For both points, I think Adaptive Cards is missing features to reference a button to certain elements in the card, or provide a way to provide additional description on the button.

Could you please do the followings?

1. [File a new bug to Adaptive Cards](https://github.com/microsoft/AdaptiveCards/issues)
2. In the bug, put a link to link it back to this bug on Web Chat
   - And optionally, put the link here to reference the bug on Adaptive Cards repo
3. Don't close this bug yet. We will keep tracking the progress on Adaptive Cards library. Once they resolved it and release a production version, Web Chat will bump to use the latest version of Adaptive Cards

**Comment by compulim** [comment]
This is resolved in Web Chat 4.15.1 which depends on AC 2.10 and it support AC schema 1.5.

AC team suggested the customer to use the `tooltip` attribute.
